### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by RWOverdijk**

### CODE REVIEW

**Comment by RWOverdijk** [comment]
The review can wait. I am changing some stuff.

**Comment by RWOverdijk** [comment]
Okay done for realsies now.

**Comment by RWOverdijk** [comment]
Okay done for realsies now.

**Comment by RWOverdijk** [comment]
Okay done for realsies now.

File: src/component/autocomplete.js [hunk 1]
```diff
@@ -212,14 +237,23 @@ export class AutoCompleteCustomElement {
...
-    this.justSelected = true;
-    this.search       = this.label(this.value);
+    result             = (arguments.length === 0) ? this.selected : result;
+    this.justSelected  = true;
+    this.value         = this.label(result);
+    this.previousValue = this.value;
+
+    this.setFocus(false);
+
+    // Because of the event we're in, databinding updates don't get called. Just skip a beat. :)
```
Comment (bas080) [inline src/component/autocomplete.js]: Aurelia has taskQueue which enables one to schedule a task. This also works but might be less idiomatic.

**Pull request merged**